src/bitmap/store.rs

```diff
@@ -43,6 +43,69 @@ impl Store {
         }
     }
 
+    pub fn insert_range(&mut self, range: Range<u16>) -> u64 {
+        // A Range is defined as being of size 0 if start >= end.
+        if range.is_empty() {
+            return 0;
+        }
+
+        match *self {
+            Array(ref mut vec) => {
+                let (range_start, range_end) = (range.start, range.end);
+
+                // Figure out the starting/ending position in the vec
+                let pos_start = vec.binary_search(&range_start).unwrap_or_else(|x| x);
+                let pos_end = vec.binary_search(&(range_end)).unwrap_or_else(|x| x);
+
+                // Overwrite the range in the middle - there's no need to take
+                // into account any existing elements between start and end, as
+                // they're all being added to the set.
+                let dropped = vec.splice(pos_start..pos_end, range);
+
+                u64::from(range_end - range_start) - dropped.len() as u64
```

Comment:
```suggestion
                u64::from(range_end - range_start) - u64::from(dropped.len())
```

Reply:
There's no `u64::from(usize)` unfortunately - I think this is because `usize` abstracts the size of the CPU native word, which in theory could be 128bits and so casting it to a u64 would be lossy.

We can use `u64::try_from(usize)` but a 128bit word doesn't seem like a real concern in practice and would require returning a `Result<u64, SomeError>` which is kinda annoying to use if the error is never really going to be triggered. 

I'm totally open to ideas to handle this better!

Reply:
Yeah, sorry, forgot about that, that's true, it is safe as there can't be more than u32 MAX value dropped! 😃